src/goto-symex/symex_goto.cpp

```diff
@@ -338,11 +338,13 @@ void goto_symext::merge_goto(const goto_statet &goto_state, statet &state)
   state.depth=std::min(state.depth, goto_state.depth);
 }
 
-void goto_symext::merge_value_sets(const goto_statet &src, statet &dest)
+void goto_symext::merge_value_sets(goto_statet &src, statet &dest)
 {
   if(dest.guard.is_false())
   {
-    dest.value_set=src.value_set;
+    // we don't actually need dest.value_set anymore, leave cleanup to later
+    // cleanup of goto states
+    dest.value_set.swap(src.value_set);
```

Comment:
Why swap and not move?

Reply:
I'll try to take a look whether adding a move operator would be as easy.

Reply:
Done (just required also default-declaring a copy constructor).